File: pubsub/src/main/java/glitch/pubsub/object/json/BitsMessage.java
package glitch.pubsub.object.json;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.time.Instant;

@Data
public class BitsMessage {
    @SerializedName("user_name")
    private final String username;
    private final String channelName;
    private final Long userId;
    private final Long channelId;
    private final Instant time;
    @SerializedName("chat_message")
    private final String message;
    private final Integer bitsUsed;
    private final Integer totalBitsUsed;
    private final String context;
}

File: pubsub/src/main/kotlin/glitch/pubsub/object/json/BitsMessage.kt
package glitch.pubsub.`object`.json

import com.google.gson.annotations.SerializedName
import java.time.Instant

data class BitsMessage(
        @SerializedName("user_name")
        val username: String, val channelName: String, val userId: Long,
        val channelId: Long, val time: Instant,
        @SerializedName("chat_message")
        val message: String, val bitsUsed: Int,
        val totalBitsUsed: Int, val context: String
)
